lib/spack/spack/config.py

```diff
@@ -363,15 +375,15 @@ def they_are(t):
         for sk, sv in iteritems(source):
             if override(sk) or sk not in dest:
                 # if sk ended with ::, or if it's new, completely override
-                dest[sk] = copy.copy(sv)
```

Comment:
I didn't check the logic, but if we need a deep copy, why aren't we using [copy.deepcopy](https://docs.python.org/2/library/copy.html)?